### DISCUSSION TITLE
Customize Product List

### DISCUSSION THREAD

**Discussion by javid-jafarov** [vendurehq/vendure#784] [General]
Hi everyone!
     I want to add one more button on product list and customize data table but i can't find code  of product list. please guide me.

**Comment by michaelbromley** [comment]
Hi!

You have 2 options here:

1. You can create your own product list view component to replace the default one. To do so, please see the guide on customizing the Admin UI: https://www.vendure.io/docs/plugins/extending-the-admin-ui/
2. If this is not feasible for whatever reason, you can also modify the source of the admin-ui project and build your own version. This is not really recommended though, since you'll then need to maintain a fork indefinitely. 

In either case, the source of the current product list component can be found here: https://github.com/vendure-ecommerce/vendure/tree/master/packages/admin-ui/src/lib/catalog/src/components/product-list

**Reply by javid-jafarov** [reply]
![2021-03-23_02-32](https://user-images.githubusercontent.com/77013349/112092492-684cb400-8bb9-11eb-9b25-8d55eb1cc191.png)
here is the attach image for my project structure, i can't find this product list folder  path
"packages/admin-ui/src/lib/catalog/src/components/product-list"
I configure this  vendure project using this link
https://www.vendure.io/docs/getting-started/
cmd is ;
yarn create @vendure my-app
please guide me.
Thank you

**Reply by javid-jafarov** [reply]
Hi!
@michaelbromley pl guide me as i have to work with cart, if i customize the product list so how should i customize the add to cart. method.
Thank you.

**Reply by michaelbromley** [reply]
Please study this section of the docs to learn how to do this: https://www.vendure.io/docs/plugins/

**Comment by javid-jafarov** [comment]
![2021-03-23_02-32](https://user-images.githubusercontent.com/77013349/112061310-205d6b00-8b80-11eb-85d8-e327a635f187.png)

here is the attach image for my project structure, i can't find this product list path
"packages/admin-ui/src/lib/catalog/src/components/product-list"
please guide me.
Thank you
